### PR TITLE
status: highlight when deps have different loaded versions

### PR DESCRIPTION
**Pull request by IanButterworth**
See https://github.com/JuliaLang/julia/pull/56769

Screenshot 2024-12-09 at 9 39 09 AM

### CODE REVIEW

File: src/Operations.jl [hunk 1]
```diff
@@ -2736,6 +2736,16 @@ function print_status(env::EnvCache, old_env::Union{Nothing,EnvCache}, registrie
                 printstyled(io, pkg_str; color=Base.warn_color())
             end
         end
+        # show if loaded and version doesn't match
+        pkg_spec = something(pkg.new, pkg.old)
+        pkgid = Base.PkgId(pkg.uuid, pkg_spec.name)
+        m = get(Base.loaded_modules, pkgid, nothing)
+        if m isa Module && pkg_spec.version !== nothing
+            v = pkgversion(m)
+            if v !== pkg_spec.version
```
Comment (KristofferC) [inline src/Operations.jl:2745]: Would using the path to the package be more robust? That would include the version but also for example if you have dvved something that happens to have the same version as the one in your main env.

Comment (IanButterworth) [reply src/Operations.jl]: Makes sense. Updated

**Comment by tecosaur** [comment]
Nit/preference: I feel like the "different version" loaded bit of the "different version loaded" is a bit redundant, just having `[loaded: v1.6.3]` in yellow would be enough IMO.

File: src/Operations.jl [hunk 2]
```diff
@@ -2736,6 +2736,25 @@ function print_status(env::EnvCache, old_env::Union{Nothing,EnvCache}, registrie
                 printstyled(io, pkg_str; color=Base.warn_color())
             end
         end
+        # show if loaded version and version in the manifest doesn't match
+        pkg_spec = something(pkg.new, pkg.old)
+        pkgid = Base.PkgId(pkg.uuid, pkg_spec.name)
+        m = get(Base.loaded_modules, pkgid, nothing)
+        if m isa Module && pkg_spec.version !== nothing
+            loaded_path = pathof(m)
+            env_path = Base.locate_package(pkgid) # nothing if not installed
```
Comment (KristofferC) [inline src/Operations.jl:2745]: There are so many ways to get the path to a package, could also for example use `pkgorigins` but this should be fine as well.

**Review by KristofferC — APPROVED** [review]

File: src/Operations.jl (hunk 2 above)
Comment (IanButterworth) [reply src/Operations.jl:2745]: `pkgorigins` only works for loaded packages? This line's the src as found based on the env, i.e. what loading would find. I thought `locate_package` would be the closest to the actual loading mechanics? If there's a more proper way then let's use it

**Pull request merged**